### PR TITLE
Diksha specific march release changes

### PR DESCRIPTION
**Pull request by aman-tunerlabs**

### CODE REVIEW

File: module/userProjects/helper.js [hunk 1]
```diff
@@ -2426,41 +2431,49 @@ module.exports = class UserProjectsHelper {
...
 
-                if( appVersion !== "" ) {
+                if (appVersion !== "") {
                     createProject["appInformation"]["appVersion"] = appVersion;
                 }
 
+                createProject["lastDownloadedAt"] = new Date();
+
+                if (data.profileInformation) {
+                    createProject.userRoleInformtion = data.profileInformation;
```
Comment (aks30) [inline module/userProjects/helper.js:2504]: @aman-tunerlabs , @deepakr123 are we storing mongo roleId in userRoleInformtion ?

File: module/userProjects/helper.js [hunk 2]
```diff
@@ -1749,569 +1763,560 @@ module.exports = class UserProjectsHelper {
...
+                        query["isAPrivateProgram"] = bodyData["filter"]["isAPrivateProgram"] = false;
+                    }
                 }
 
-                query = {...query, ...filterQuery[0]};
-                
-            }
+                let projects = await this.projects(
+                    query,
+                    CONSTANTS.common.DEFAULT_PAGE_SIZE,
```
Comment (aks30) [inline module/userProjects/helper.js:1930]: @aman-tunerlabs what is someone whats an another value for page size and not the default one?

File: module/userProjects/helper.js (hunk 1 above)
Comment (aman-tunerlabs) [reply module/userProjects/helper.js:2504]: @deepakr123  Can you confirm on this please ?

File: module/userProjects/helper.js (hunk 2 above)
Comment (aman-tunerlabs) [reply module/userProjects/helper.js:1930]: @aks30  This logic works for pagination having search field and page number . Pagination in this api should work not just for user assigned project but should also work for the one that is targeted. So Fetching all the assigned project and combining the targeted ones and finally doing pagination on the merged data.

Comment (aman-tunerlabs) [reply module/userProjects/helper.js:1930]: @aks30  Current apis works with pagination so no issue on that.

**Pull request merged**